protocol/bc/legacy/txinput.go

```diff
@@ -148,42 +148,44 @@ func (t *TxInput) readFrom(r io.Reader) (err error) {
 	}
 
 	t.WitnessSuffix, _, err = blockchain.ReadExtensibleString(r, func(r io.Reader) error {
-		// TODO(bobg): test that serialization flags include SerWitness, when we relax the serflags-must-be-0x7 rule
-		if ii != nil {
-			// read IssuanceInput witness
-			_, err = ii.InitialBlock.ReadFrom(r)
-			if err != nil {
-				return err
-			}
+		if t.AssetVersion == 1 {
```

Comment:
Maybe invert this test and return early? It is pretty deeply nested and this seems like an "error case".